### ISSUE TITLE
Network with skip connections

### ISSUE THREAD

**Issue by Deathn0t** [tensorspace-team/tensorspace#119]
Hello,

I would like to build this network with tensorspace, is it possible ?

![model](https://user-images.githubusercontent.com/5201978/48644628-d0b37700-e9a8-11e8-9234-c60a024e6b90.png)

Here is the code I tried:

```
{% load static %}
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>TensorSpace - Saddle</title>
    <meta name="author" content="syt123450 / https://github.com/syt123450">

    <script src="{% static "lib/three.min.js" %}"></script>
    <script src="{% static "lib/tween.min.js" %}"></script>
    <script src="{% static "lib/tf.min.js" %}"></script>
    <script src="{% static "lib/TrackballControls.js" %}"></script>
    <script src="{% static "lib/tensorspace.min.js" %}"></script>

    <script src="{% static "lib/jquery.min.js" %}"></script>

    <style>

        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        #container {
            width: 100%;
            height: 100%;
        }
    </style>

</head>
<body>

<div id="container"></div>

<script>

    $(function() {

		let data_path = "{% static "saddle/" %}";
		let modelContainer = document.getElementById( "container" );

		let input_1 = new TSP.layers.Input1d({
			shape: [10]
		});

		let dense_1 = new TSP.layers.Dense({ units: 80 });
		dense_1.apply(input_1);

		let dropout_1 = new TSP.layers.Layer1d( { shape: [80] } );
		dropout_1.apply(dense_1);

		let dense_2 = new TSP.layers.Dense({ units: 5 });
		dense_2.apply(dropout_1);

		let dropout_2 = new TSP.layers.Layer1d( { shape: [5] } );
		dropout_2.apply(dense_2);

		let dropout_3 = new TSP.layers.Layer1d( { shape: [10] } );
		dropout_3.apply(input_1);

		let dense_3 = new TSP.layers.Dense({ units: 160 });
		dense_3.apply(dropout_3);

		let dropout_4 = new TSP.layers.Layer1d({ shape: [160] });
		dropout_4.apply(dense_3);

		let dense = new TSP.layers.Dense({ units: 20 });
		dense.apply(input_1);

		let dropout = new TSP.layers.Layer1d({ shape: [20]});
		dropout.apply(dense);

		let concatenate = new TSP.layers.Layer1d( { shape: [185] } );
		concatenate.apply(dropout_4);

		let dense_4 = new TSP.layers.Dense({ units: 1 });
		dense_4.apply(concatenate);

		let model = new TSP.models.Model(modelContainer, {
			inputs: [input_1],
			outputs: [dense_4],
		});

		console.log("ici");

		model.load({
			type: "tfjs",
			url: data_path+"model/model.json",
			onComplete: function() {
				console.log( "\"Hello World!\" from TensorSpace Loader." );
			}
		});

		model.init( function() {

			$.ajax({
				url: data_path+"data/s1.json",
				type: 'GET',
				async: true,
				dataType: 'json',
				success: function (data) {

					model.predict( data );

				}
			});


		} );

    });

</script>

</body>
</html>
```

and here is the result I have

![screenshot 2018-11-16 at 13 48 18](https://user-images.githubusercontent.com/5201978/48644738-2ab43c80-e9a9-11e8-8628-6be4bee0f4cd.png)

Thank you for your help !

**Comment by syt123450** [comment]
TensorSpace can be used to create skip connection such as Inception block in Inception or residual block in ResNet. However, the merge function in TensorSpace only support for 3d layer (https://tensorspace.org/html/docs/mergeConcate.html) now.

Temporarily, TensorSpace can not create skip connection for 1d layers(like dense layer), I am working it, and will add this new feature to TensorSpace as soon as possible.

**Comment by Deathn0t** [comment]
@syt123450 thanks ! that would be very helpful !

**Comment by syt123450** [comment]
Hi @Deathn0t , the latest TensorSpace ( [ v0.2.0 ](https://github.com/tensorspace-team/tensorspace/releases/tag/v0.2) ) supports 1d concatenate operation. For your skip connection graph, I wrote a code sample in [ this repo ](https://github.com/syt123450/skip-connection-1d). In this sample, I generated a random model using TensorFlow.js ( [trainingModel.html](https://github.com/syt123450/skip-connection-1d/blob/master/trainingModel.html) ) and visualized it with TensorSpace ( [index.html](https://github.com/syt123450/skip-connection-1d/blob/master/index.html) ).

However, as Dropout layer only take effect in model training process, I did not add it to TensorSpace model. If you want to add it to your TensorSpace model, just like your previous code, use Layer1d to replace the Dropout Layer.

Hope these code snippets can help ~

**Comment by Deathn0t** [comment]
Thank you @syt123450 ! I will try it soon !
